### PR TITLE
Reset course project

### LINKED ISSUE #663: Changing/undoing course selection
#489

### PR DESCRIPTION
**Pull request by jaakkonarhi**
# Description of the PR

+ Action for removing the current course in the A+ menu
+ Closes #663 

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>e2e</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [x] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [x] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>e2e</b> tests created</li>
        <li>- [ ] all <b>e2e</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/intellij-plugin/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

File: src/main/resources/resources.properties [hunk 1]
```diff
@@ -207,4 +207,6 @@ ui.tutorial.TutorialActionGroup.step=Step {0}
 
 ### Misc
 ui.localHistory.submission.tag={0}, {1}, Submission {2}
-ui.module.ModuleListView.turnIntoAPlusProject=<html><div style='text-align: center;'>Click here to turn this project into an A+ project.<br>(or go to: A+ | Turn Project Into A+ Project)</div></html>
\ No newline at end of file
+ui.module.ModuleListView.turnIntoAPlusProject=<html><div style='text-align: center;'>Click here to turn this project into an A+ project.<br>(or go to: A+ | Turn Project Into A+ Project)</div></html>
+ui.ResetProjectAction.confirmDialog.title=Change Course
```
Comment (OlliKiljunen) [inline src/main/resources/resources.properties:211]: This perhaps should be something else.

File: src/main/java/fi/aalto/cs/apluscourses/intellij/actions/ResetProjectAction.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.DumbAwareAction;
+import com.intellij.openapi.project.ProjectManager;
+import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
+import javax.swing.JOptionPane;
+import org.jetbrains.annotations.NotNull;
+
+class ResetProjectAction extends DumbAwareAction {
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/intellij/actions/ResetProjectAction.java:12]: Should this be only available if it's a course project?

File: src/main/java/fi/aalto/cs/apluscourses/intellij/actions/ResetProjectAction.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    if (JOptionPane.showConfirmDialog(null,
+        getText("ui.ResetProjectAction.confirmDialog.content"),
+        getText("ui.ResetProjectAction.confirmDialog.title"),
+        JOptionPane.OK_CANCEL_OPTION) == JOptionPane.OK_OPTION) {
+
+      PluginSettings.getInstance().getCourseFileManager(project).delete();
+
+      ProjectManager.getInstance().reloadProject(project);
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/intellij/actions/ResetProjectAction.java:28]: I'm wondering if some concurrent action can possibly re-create the project file between these two statements. But that's probably a rare occurrence without any severe consequences so it doesn't matter.

**Review by OlliKiljunen — APPROVED** [review]
Good! 👍 A couple of comments about UI issues but code is merge-ready.

**Review by stellatsv — APPROVED** [review]

**Pull request merged**